unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIComponentsUtils/UIEventSubscription.cs

```diff
@@ -0,0 +1,71 @@
+using DCL.Controllers;
+using DCL.ECS7.InternalComponents;
+using DCL.Models;
+using System;
+using System.Linq;
+using UnityEngine.UIElements;
+
+namespace DCL.ECSComponents
+{
+    public class UIEventSubscription<TEvent> : IDisposable where TEvent : PointerEventBase<TEvent>, new()
+    {
+        private readonly VisualElement uiElement;
+        private readonly IParcelScene parcelScene;
+        private readonly IDCLEntity dclEntity;
+        private readonly PointerEventType pointerEventType;
+        private readonly InputAction inputAction;
+        private readonly IInternalECSComponent<InternalInputEventResults> results;
+
+        public UIEventSubscription(VisualElement uiElement,
+            IParcelScene parcelScene,
+            IDCLEntity dclEntity,
+            PointerEventType pointerEventType,
+            PBPointerEvents.Types.Info requestInfo,
+            IInternalECSComponent<InternalInputEventResults> results
+        )
+        {
+            this.uiElement = uiElement;
+            this.parcelScene = parcelScene;
+            this.dclEntity = dclEntity;
+            this.pointerEventType = pointerEventType;
+            this.results = results;
+
+            // every other field is irrelevant for UI
+            inputAction = requestInfo.Button;
+
+            uiElement.RegisterCallback<TEvent>(OnEventRaised);
+        }
+
+        private void OnEventRaised(TEvent evt)
+        {
+            // filter by button
+            if (!IsInputAccepted(evt))
+                return;
+
+            var eventData = new InternalInputEventResults.EventData
+            {
+                button = (InputAction) evt.button,
+                analog = 1,
```

Comment:
Should this be analog 0?

Reply:
I believe it's irrelevant for UI, so I can just omit this field